Horde3D/Source/Horde3DEngine/egTexture.cpp

```diff
@@ -228,18 +228,11 @@ TextureResource::TextureResource( const string &name, uint32 width, uint32 heigh
 	}
 	else
 	{
-		uint32 size = rdi->calcTextureSize( _texFormat, width, height, depth );
-		unsigned char *pixels = new unsigned char[size];
-		memset( pixels, 0, size );
-		
 		_texType = flags & ResourceFlags::TexCubemap ? TextureTypes::TexCube : TextureTypes::Tex2D;
 		if( depth > 1 ) _texType = TextureTypes::Tex3D;
 		_sRGB = (_flags & ResourceFlags::TexSRGB) != 0;
 		_texObject = rdi->createTexture( _texType, _width, _height, _depth, _texFormat,
 		                                 _maxMipLevel, _maxMipLevel > 0, false, _sRGB );
-		rdi->uploadTextureData( _texObject, 0, 0, pixels );
-		
```

Comment:
Why was it deleted? Is it guaranteed by OpenGL and GLES specification that texture will be cleared and that there will be no garbage?
If yes then no problem, otherwise it can lead to problems on some implementations.

Reply:
Aside from the texture question, everything seems to be fine. Thanks for your changes. As for GLES backend, I'll test it and see, whether texture border could be implemented there.

Reply:
I've checked the spec, glTexImage2D is not required to clear the memory if it is given null pointer. And it is undefined behaviour if this texture is attached to geometry. Therefore, it will cause undefined behaviour for default textures (2d, cube, 3d) that are attached to mesh when no texture is defined.

https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glTexImage2D.xhtml